packages/starlight/__tests__/basics/config-errors.test.ts

```diff
@@ -253,3 +253,32 @@ test('errors with sidebar entry that includes `items` and `autogenerate`', () =>
 			**sidebar.0**: Unrecognized key(s) in object: 'autogenerate'"
 	`);
 });
+
+test('parses route middleware config successfully', () => {
+	const data = parseStarlightConfigWithFriendlyErrors({
+		title: '',
+		routeMiddleware: './src/routeData.ts',
+	});
+	expect(data.routeMiddleware).toEqual(['./src/routeData.ts']);
+});
+
+test("errors if one route middleware is conflicting path with Astro' middleware", () => {
+	expect(() =>
+		parseStarlightConfigWithFriendlyErrors({
+			title: 'Test',
+			routeMiddleware: ['src/middleware.ts', 'src/routeData.ts'],
+		})
+	).toThrowErrorMatchingInlineSnapshot(
+		`
+		"[AstroUserError]:
+			Invalid config passed to starlight integration
+		Hint:
+			A \`routeMiddleware\` file path in your Starlight config conflicts with Astro’s middleware locations. It cannot be one of the following: "src/middleware.js", "src/middleware.ts", "src/middleware/index.js", "src/middleware/index.ts".
+			
+			You should move your Starlight route middleware file \`src/middleware.ts\` somewhere else like \`src/starlightRouteData.ts\` and update the \`routeMiddleware\` file path to match.
```

Comment:
And the error message will need updating too:

```suggestion
			The \`"./src/middleware.ts"\` path in your Starlight \`routeMiddleware\` config conflicts with Astro’s middleware locations.
			
			You should rename \`./src/middleware.ts\` to something else like \`./src/starlightRouteData.ts\` and update the \`routeMiddleware\` file path to match.
```